components/nginx/scripts/preconfigure.py

```diff
@@ -118,14 +151,34 @@ def preconfigure_nginx():
 
 def create_certs():
     utils.mkdir(utils.SSL_CERTS_TARGET_DIR)
-    utils.generate_ca_cert()
+    ca_cert_deployed, ca_key_deployed = _deploy_cert_and_key(
+        'ca_certificate', 'ca_key',
+        utils.INTERNAL_CA_CERT_PATH, utils.INTERNAL_CA_KEY_PATH)
+    has_ca_key = ca_key_deployed
+    if not ca_cert_deployed:
```

Comment:
This is getting a bit convoluted. Let's add some comments here. I know the code should be readable enough by itself, but I'm afraid future generations will have trouble with it.

Reply:
added a few comments (didn't affect this very line so github won't hide this message)